### PR TITLE
Allow spaces inside of unquoted keys

### LINKED ISSUE #86: Spaces in Unquoted Block Keys
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Spaces in the middle of unquoted block keys breaks syntax highlighting.

### Steps to Reproduce

1. Add an unquoted key for a block containing a space in the middle
    - Example: `Space Ghost: >`
    - Example: `Ghost in the Shell: |`

**Expected behavior:** The key should become red like a key.

**Actual behavior:** The key remains green like a value.

**Reproduces how often:** Consistent (100%)

### Versions

 - Atom 1.15.0 (x64)
 - language-yaml 0.28.0

### Additional Information

The CSS classes should be `syntax--entity syntax--name syntax--tag`, but are `syntax--string syntax--unquoted`.

### PR DESCRIPTION
**Pull request by deckar01**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change

The YAML spec allows spaces and tabs inside of a unquoted block key, but this grammar is checking for `\S` (non-whitespace). Checking for `.` instead allows tabs and spaces while continuing to exclude line breaking whitespace.

### Alternate Designs

I considered using the more explicit `[^\n\r\f]`, but `.` is more concise and widely understood to not match line breaking escape codes.

### Benefits

Better compliance with the YAML spec and correct syntax highlighting.

### Possible Drawbacks

None

### Applicable Issues

Closes #86

### CODE REVIEW

**Comment by 50Wliu** [comment]
Can there be one extra test to ensure that in something like `Ghost in the Shell  : |` the whitespace between `Shell` and `:` is not tokenized as `entity.name.tag.yaml`?

**Comment by deckar01** [comment]
Good catch. It was incorrectly capturing trailing spaces.

File: grammars/yaml.cson [hunk 1]
```diff
@@ -38,7 +38,7 @@
   {
     # hello: >
     # hello: |
-    'begin': '^(\\s*)(?!-\\s*)(\\S+)\\s*(:)(?:\\s+((!)[^!\\s]+))?\\s+(?=\\||>)'
+    'begin': '^(\\s*)(?!-\\s*)(.*\\S)\\s*(:)(?:\\s+((!)[^!\\s]+))?\\s+(?=\\||>)'
```
Comment (deckar01) [inline grammars/yaml.cson]: The spec shows `( s-white* ns-plain-char(c) )*` which is effectively `((?:[ \t]*\S)*)`.
Note that `((?:[ \t]*\S)*)` = `((?:[ \t\S]*\S)*)` = `((?:.*\S)*)` = `((?:.*\S)*)`.
Since `(\S+)` already ignores the empty string case, `((?:.*\S)*)` ≈ `(.*\S)`.

I think `(.*\S)` is easier to read an understand than `((?:[ \t]*\S)*)` and the intermediate expressions.

**Comment by 50Wliu** [comment]
Thanks!
